manila-cli.py

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python
+
+# Copyright (c) 2015 Scality
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import grp
+import os
+import pwd
+
+from scality_manila_utils.helper import Helper
+
+
+def drop_privileges():
+    """
+    Attempt to drop privileges.
+    """
+    def _try_get(f, *args, **kwargs):
+        try:
+            return f(*args, **kwargs)
+        except KeyError:
+            return None
+
+    def get_user(name):
+        return _try_get(pwd.getpwnam, name)
+
+    def get_group(name):
+        return _try_get(grp.getgrnam, name)
+
+    # Attempt to find a proper user and group
+    user_info = get_user('nobody')
+    group_info = get_group('nogroup')
+    if group_info is None:
+        group_info = get_group('nobody')
+
+    # Drop privileges
+    if user_info and group_info:
+        os.setegid(group_info.gr_gid)
+        os.seteuid(user_info.pw_uid)
+
+
+if __name__ == '__main__':
+    if os.geteuid() != 0:
+        raise RuntimeError("This program requires superuser privileges")
+
+    # Drop any elevated permissions
+    drop_privileges()
+
+    pre_parser = argparse.ArgumentParser(add_help=False)
+    pre_parser.add_argument('--exports', help='Path to exports file',
+                            default='/etc/exports.conf')
+    pre_parser.add_argument('--basedir', help='Directory where manila '
```

Comment:
Is this required? The script will need to NFS-mount `localhost:/`, so it could create a temporary directory to do so (which removes potential concurrency issues).